<a href="https://colab.research.google.com/github/kotamokaku/game-analysis-log/blob/main/day1_game_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

# 1. 架空のデータを作る（ユーザーID、プレイ時間、課金有無）
data = {
    'user_id': [1, 2, 3, 4, 5],
    'play_time_min': [120, 15, 250, 45, 100],
    'is_charged': [True, False, True, False, False]
}

# 2. Pandasの「データフレーム」という形に変換する
df = pd.DataFrame(data)

# 3. 中身を確認する (head)
print("--- データの先頭を確認 ---")
print(df.head())

# 4. 統計的な要約を確認する (describe)
print("\n--- データの統計的な特徴を確認 ---")
print(df.describe())

# 5. 特定の条件（課金ユーザーだけ）で絞り込む
charged_users = df[df['is_charged'] == True]
print("\n--- 課金ユーザーのみの抽出 ---")
print(charged_users)

--- データの先頭を確認 ---
   user_id  play_time_min  is_charged
0        1            120        True
1        2             15       False
2        3            250        True
3        4             45       False
4        5            100       False

--- データの統計的な特徴を確認 ---
        user_id  play_time_min
count  5.000000       5.000000
mean   3.000000     106.000000
std    1.581139      90.788215
min    1.000000      15.000000
25%    2.000000      45.000000
50%    3.000000     100.000000
75%    4.000000     120.000000
max    5.000000     250.000000

--- 課金ユーザーのみの抽出 ---
   user_id  play_time_min  is_charged
0        1            120        True
2        3            250        True


#### このデータについての第一印象
- **構成**: 行数は5件、列は3個（user_id, play_time_min, is_charged）。
- **属性**: 数値データは play_time_min、カテゴリデータは is_charged。
- **違和感**: プレイ時間の平均は106分だが、最大値（250分）が平均の2倍以上。一部のヘビーユーザーが平均を押し上げている。
- **分析できそうなこと**: 課金の有無によって、プレイ時間にどれくらい明確な差が出るのかを比較したい。

#### Day2：仮説
- **仮説**: 「課金をしているプレイヤーほど、プレイ時間が長い傾向にある」